H/HiGHS/HiGHSstatic/build_tarballs.jl

```diff
@@ -0,0 +1,25 @@
+
+include("../highs_common.jl")
+
+script = build_script(shared_libs = "OFF")
+
+products = [
+    ExecutableProduct("highs", :highs),
+]
+
+dependencies = [
+    BuildDependency("CompilerSupportLibraries_jll"),
```

Comment:
[This](https://github.com/JuliaPackaging/Yggdrasil/pull/4456#issuecomment-1042483581) also means that this is still a runtime dependency.  I'm not sure this has achieved what you were hoping :confused:

Reply:
No, it didn't :cry:

What I want is a single executable that someone copy onto a Windows machine, assuming they have probably never programmed before and have no compilers or support libraries installed.

The main problem is that I don't have a windows machine to test, and @jajhall has one, but his test probably worked because he had some things already installed.

Reply:
There is the possibility we can do a fully static build by using static libraries of GCC (might need some manual trickeries), but also note that the binary distribution will basically be GPL-licensed.

Reply:
I'm not sure we want to got the GPL route. 

An easier option might be that we just manually curate a download for windows users and host that at HiGHS. I think even the current workflow of getting Windows users to download a file called `HiGHS.v1.2.0.x86_64-w64-mingw32-cxx11.tar.gz` is probably a little intimidating.

Let's leave this for now, and I'll have a chat with @jajhall about our next steps.

Reply:
For the record, it's sufficient to do
```diff
diff --git a/H/HiGHS/highs_common.jl b/H/HiGHS/highs_common.jl
index 8e6fd4e50..ca0e8d734 100644
--- a/H/HiGHS/highs_common.jl
+++ b/H/HiGHS/highs_common.jl
@@ -31,6 +31,9 @@ fi
 mkdir -p HiGHS/build
 cd HiGHS/build
 
+if [[ "${BUILD_SHARED}" == "OFF" ]]; then
+    export CXXFLAGS="-static"
+fi
 cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
     -DCMAKE_BUILD_TYPE=Release \
```
to get a truly static library (in terms of GCC dependencies):
```console
sandbox:${WORKSPACE}/srcdir/HiGHS/build # ${target}-objdump -p ${bindir}/highs.exe | grep DLL
 vma:            Hint    Time      Forward  DLL       First
        DLL Name: KERNEL32.dll
        DLL Name: msvcrt.dll
sandbox:${WORKSPACE}/srcdir/HiGHS/build # ls -lh ${bindir}/highs.exe
-rwxr-xr-x    1 root     root       16.4M Feb 17 02:07 /workspace/destdir/bin/highs.exe
```
The cost of having a 16 MiB executable.  And again, you're embedding GCC.  However, being a standalone executable, the GPL license may not actually be a big deal: you don't need to link to anything else, so it doesn't "spread".  Of course IANAL, blah blah.

Reply:
I think a benefit of doing the static build is that you don't need to expand the C++ string ABI (that's needed to match user's `libstdc++`), so `HiGHS.v1.2.0.x86_64-w64-mingw32-cxx11.tar.gz` will simplify to `HiGHS.v1.2.0.x86_64-w64-mingw32.tar.gz` :slightly_smiling_face:

Reply:
Okay, nice. Perhaps providing a GPL binary for general users would be okay, and their commercial customers can pay for support if they don't want to use it.

I'll let @jajhall and @lgottwald make the decision since it's not really up to me.

Reply:
> Perhaps providing a GPL binary for general users would be okay, and their commercial customers can pay for support if they don't want to use it.

This seems reasonable to me. After all, it's far from being the only way to use HiGHS, even via binaries. It's just an option.

I've gone back to the static binary that @odow produced, and it only ran because I put it in the same folder as the support binaries. Removing them, and even adding the two demanded by the pop-up gives the same behaviour as when I first used the shared binary.

Thanks for your efforts @odow and @giordano

Reply:
@jajhall can you please try https://julia-bb-buildcache.s3.amazonaws.com/69df64b00a9ef2a2a4492e0832af484997dfeb151f8e1a9cb2f525528950c9d1/648abe945f5b43151bd5a1c1fb098ccb13c71bf620c8d675ef95128087800780/x86_64-w64-mingw32-cxx11.tar.gz?